Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+                xmlqname = new javax.xml.namespace.QName(CoreX12AuditDataTransformConstants.CORE_X12_NAMESPACE_URI, CoreX12AuditDataTransformConstants.CORE_X12_REQUEST_LOCALPART);
+                JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest> element = new JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest>(xmlqname, COREEnvelopeRealTimeRequest.class, oRequestNoPayload);
+                marshaller.marshal(element, baOutStrm);
+                bObject = baOutStrm.toByteArray();
+            }
+
+            if (msg instanceof COREEnvelopeRealTimeResponse) {
+                oResponseNoPayload = (COREEnvelopeRealTimeResponse) msg;
+                oResponseNoPayload.setPayload("");
+                xmlqname = new javax.xml.namespace.QName(CoreX12AuditDataTransformConstants.CORE_X12_NAMESPACE_URI, CoreX12AuditDataTransformConstants.CORE_X12_RESPONSE_LOCALPART);
+                JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse> element = new JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse>(xmlqname, COREEnvelopeRealTimeResponse.class, oResponseNoPayload);
+                marshaller.marshal(element, baOutStrm);
+                bObject = baOutStrm.toByteArray();
+            }
+
+            if (msg instanceof COREEnvelopeBatchSubmission) {
+                oBatchRequestNoPayload = (COREEnvelopeBatchSubmission) msg;
+                oBatchRequestNoPayload.setPayloadID("");
+                xmlqname = new javax.xml.namespace.QName(CoreX12AuditDataTransformConstants.CORE_X12_NAMESPACE_URI, CoreX12AuditDataTransformConstants.CORE_X12_BATCH_REQUEST_LOCALPART);
+                JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmission> element = new JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmission>(xmlqname, COREEnvelopeBatchSubmission.class, oBatchRequestNoPayload);
+                marshaller.marshal(element, baOutStrm);
+                bObject = baOutStrm.toByteArray();
+            }
+
+            if (msg instanceof COREEnvelopeBatchSubmissionResponse) {
+                oBatchResponseNoPayload = (COREEnvelopeBatchSubmissionResponse) msg;
+                oBatchResponseNoPayload.setPayloadID("");
+                xmlqname = new javax.xml.namespace.QName(CoreX12AuditDataTransformConstants.CORE_X12_NAMESPACE_URI, CoreX12AuditDataTransformConstants.CORE_X12_BATCH_RESPONSE_LOCALPART);
+                JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmissionResponse> element = new JAXBElement<org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmissionResponse>(xmlqname, COREEnvelopeBatchSubmissionResponse.class, oBatchResponseNoPayload);
+                marshaller.marshal(element, baOutStrm);
+                bObject = baOutStrm.toByteArray();
+            }
+        } catch (JAXBException ex) {
+            throw new RuntimeException(ex.getCause());
```

Comment:
do we want to throw a runtime exception for audit logging?  won't this kill the whole transaction.  auditing shouldn't cause an inbound/outbound message to fail.